identity_stardust/src/stardust_document.rs

```diff
@@ -51,8 +50,52 @@ impl StardustDocument {
     )
   }
 
+  /// Temporary testing implementation.
+  pub fn tmp_add_verification_method(
+    &mut self,
+    id: CoreDID,
+    keypair: &KeyPair,
+    fragment: &str,
+    scope: MethodScope,
+  ) -> Result<()> {
+    let method: VerificationMethod = VerificationMethod::new(id, keypair.type_(), keypair.public(), fragment)?;
+
+    self.0.insert_method(method, scope)?;
+
+    Ok(())
+  }
+
+  /// Temporary testing implementation.
+  pub fn tmp_add_service(&mut self, id: CoreDID, fragment: &str, type_: &str, endpoint: ServiceEndpoint) -> Result<()> {
+    let service = Service::builder(Object::new())
+      .type_(type_)
+      .id(id.join(fragment).unwrap())
+      .service_endpoint(endpoint)
+      .build()?;
+
+    self.0.service_mut().append(service);
+
+    Ok(())
+  }
+
+  /// Temporary testing implementation.
+  pub fn tmp_set_id(&mut self, mut id: CoreDID) {
+    std::mem::swap(self.0.id_mut(), &mut id);
+  }
+
+  /// Temporary testing implementation.
+  pub fn tmp_id(&self) -> &CoreDID {
+    self.0.id()
+  }
+
+  /// Serializes the document for inclusion in an alias output's state metadata.
+  pub fn into_state_metadata_bytes(self, encoding: StateMetadataEncoding) -> Result<Vec<u8>> {
+    let state_metadata_doc = StateMetadataDocument::from(self);
+    state_metadata_doc.pack(encoding)
+  }
```

Comment:
This should probably just be `pack`, and `encoding` should be optional or moved to another function? Doesn't matter too much right now, will need an API pass later.

Aside: it would be nice to be able to use the [packable](https://docs.rs/packable/latest/packable/trait.Packable.html) like `bee`/`iota.rs` does, but that's beside the point.